src/main.ts

```diff
@@ -109,27 +118,55 @@ function setColorRampDisabled(disabled: boolean): void {
 
 // DATASET LOADING ///////////////////////////////////////////////////////
 
+let collection: string;
+let collectionData: Map<string, CollectionEntry>;
 let dataset: Dataset | null = null;
 let datasetName = "";
 let datasetOpen = false;
 let featureName = "";
 let selectedTrack: Track | null = null;
-// TODO: Get the first dataset in a manifest JSON?
-const DEFAULT_DATASET = "mama_bear";
 
 async function loadDataset(name: string): Promise<void> {
   console.time("loadDataset");
   datasetOpen = false;
+  const hasLoadedOtherDataset: boolean = dataset !== null && datasetName !== name;
+  const lastDataset = datasetName;
   datasetSelectEl.disabled = true;
   featureSelectEl.disabled = true;
 
   if (dataset !== null) {
     dataset.dispose();
   }
 
+  if (collectionData && !collectionData.has(name)) {
+    console.warn(`Collection does not include '${name}' as a dataset. Defaulting to first dataset in the collection.`);
+    name = getDefaultDatasetName(collectionData);
+  }
+
   datasetName = name;
-  dataset = new Dataset(`${baseUrl}/${name}`);
-  await dataset.open();
+  datasetSelectEl.value = name;
+  try {
+    let datasetPath;
+    if (isUrl(name)) {
+      datasetPath = getDatasetPath(name);
+    } else {
+      datasetPath = getDatasetPathFromCollection(name, collection, collectionData);
+    }
```

Comment:
it looks like lines 149-154 should be the body of just one function call to a function exported from UrlUtility